### PR TITLE
Fix cosmic HI fraction plot

### PR DESCRIPTION
**Pull request by MatthieuSchaller**
Three changes:
 - Remove Walter et al. 2020 data
 - Correct the simulations for He abundance
 - Add cosmology correction from Schaye+01

### CODE REVIEW

File: colibre/scripts/HI_mass_evolution.py [hunk 1]
```diff
@@ -68,49 +69,59 @@
...
 P20_rhoHI_lo = P20eq13(zgrid, *P20_data[1])
 P20_rhoHI_hi = P20eq13(zgrid, *P20_data[2])
+
+# Cosmology correction (Schaye 2001)
+P20_rhoHI *= comso.h ** -1 * sqrt(cosmo.Om0 * (1.0 + zgrid) ** 3 + comso.Ode0)
+P20_rhoHI /= 0.7 ** -1 * sqrt(0.3 * (1.0 + zgrid) ** 3 + 0.7)
+P20_rhoHI_lo *= comso.h ** -1 * sqrt(cosmo.Om0 * (1.0 + zgrid) ** 3 + comso.Ode0)
+P20_rhoHI_lo /= 0.7 ** -1 * sqrt(0.3 * (1.0 + zgrid) ** 3 + 0.7)
+P20_rhoHI_hi *= comso.h ** -1 * sqrt(cosmo.Om0 * (1.0 + zgrid) ** 3 + comso.Ode0)
+P20_rhoHI_hi /= 0.7 ** -1 * sqrt(0.3 * (1.0 + zgrid) ** 3 + 0.7)
```
Comment (EvgeniiChaikin) [inline colibre/scripts/HI_mass_evolution.py:90]: I was running this branch and found a couple of typos. Below is a suggested change to resolve the typos.

```suggestion
P20_rhoHI *= cosmo.h ** -1 * np.sqrt(cosmo.Om0 * (1.0 + zgrid) ** 3 + cosmo.Ode0)
P20_rhoHI /= 0.7 ** -1 * np.sqrt(0.3 * (1.0 + zgrid) ** 3 + 0.7)
P20_rhoHI_lo *= cosmo.h ** -1 * np.sqrt(cosmo.Om0 * (1.0 + zgrid) ** 3 + cosmo.Ode0)
P20_rhoHI_lo /= 0.7 ** -1 * np.sqrt(0.3 * (1.0 + zgrid) ** 3 + 0.7)
P20_rhoHI_hi *= cosmo.h ** -1 * np.sqrt(cosmo.Om0 * (1.0 + zgrid) ** 3 + cosmo.Ode0)
P20_rhoHI_hi /= 0.7 ** -1 * np.sqrt(0.3 * (1.0 + zgrid) ** 3 + 0.7)
```

**Review by bwvdnbro — CHANGES_REQUESTED** [review]
I think the helium correction that is mentioned in the caption and the script is a bit misleading; I had to read through Peroux & Howk (2020) to understand what is happening.
The fit in Peroux & Howk (2020) was made to the _neutral gas density_, which assumes a gas that consists of hydrogen and helium. However, this plot claims to show the _HI mass density_, so not including helium. Either the plot quantity needs to be renamed in the name of the plot and the labels, or the helium correction should be performed on the data and not on the simulation. Either way, it should not be called a _correction_; it is a _conversion factor_ because of a change of quantity.

File: colibre/scripts/HI_mass_evolution.py [hunk 2]
```diff
@@ -60,6 +60,7 @@
     redshift = data.z
     HI_mass = data.gas_hi_mass.to("Msun")
     HI_mass_density = HI_mass / box_volume
+    HI_mass_density *= 1.0 / 0.76  # Apply Helium correction
```
[the remark this answers is not in the document]
Comment (MatthieuSchaller) [reply colibre/scripts/HI_mass_evolution.py:63]: Just saw I never replied. 

Agreed, my comment is not good enough. I am happy with either of your propositions. Feel free to make the changes in a way that makes everything clear for you also to maintain in the long run.

**Comment by bwvdnbro** [comment]
Do we want to keep the commented out sections in the script?

**Review by bwvdnbro — APPROVED** [review]

**Comment by MatthieuSchaller** [comment]
We could if we want to have that other dataset reappear in the future.

**Comment by bwvdnbro** [comment]
What would be needed to make that dataset reappear? I'm happy to keep it like this, but it would be useful to provide a comment explaining why it was disabled and what would be needed to make it reappear.

**Comment by MatthieuSchaller** [comment]
We are now correcting the simulation to match the Peroux dataset. 

The Walter dataset could reappear if the same correction (He) was applied.

**Comment by MatthieuSchaller** [comment]
@jtrayford, I think you added the Walter data for the cosmic HI. Do you know whether they applied any He correction? Or any other correction?

**Comment by bwvdnbro** [comment]
From the paper, section 3.2.2:
> However, we do account for the contribution of helium, which corresponds to a correction factor of μ = 1.3.

So the same correction as for the Peroux & Howk data.

**Comment by bwvdnbro** [comment]
Do we also need the cosmology correction that was applied to the Peroux & Howk data? And where does this come from (Schaye 2001 is not a particularly clear reference - there are 4 of those)?

**Comment by MatthieuSchaller** [comment]
It's eq. 14-15 of https://ui.adsabs.harvard.edu/abs/2001ApJ...559..507S/abstract

Yes, it would need to be applied as well. On top of the changes to rho_crit.

**Comment by bwvdnbro** [comment]
I think everything is ready now. The plot now looks like this (the simulation is a 6 Mpc low resolution box):
HI_mass_evolution

**Comment by MatthieuSchaller** [comment]
Thanks!

**Comment by MatthieuSchaller** [comment]
Approved.

**Pull request merged**